modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusResources.scala

```diff
@@ -1,14 +1,18 @@
 package org.tessellation.sdk.infrastructure.consensus
 
 import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.infrastructure.consensus.declaration.kind.PeerDeclarationKind
 import org.tessellation.security.hash.Hash
 
+/** Represents various data received from other peers
+  */
 case class ConsensusResources[A](
   peerDeclarationsMap: Map[PeerId, PeerDeclarations],
-  artifacts: Map[Hash, A],
-  removedFacilitators: Set[PeerId]
+  acksMap: Map[(PeerId, PeerDeclarationKind), Set[PeerId]],
+  ackKinds: Set[PeerDeclarationKind],
```

Comment:
isn't `ackKinds` just a `Set` of all `PeerDeclarationKind` from `acksMap`? If so then we are duplicating data here so maybe some flattening function will be better 🤔

Reply:
It is, but it's an optimization. Duplication (denormalization) of information is a common technique to optimize access to data.

There's almost no additional cost for writing to this Set, but there's a significant benefit on reading, and reading is as frequent as writing is (hundreds or even thousands of times during a consensus).